File: hw/fuentes/U2/HW10SaveJson/VinylRecord/src/ec/edu/espe/model/VinylRecord.java

package ec.edu.espe.model;

/**
 *
 * @author Jorge Fuentes
 */
public class VinylRecord {
    String title;
    String artist;
    int year;
    String gener;
    String duration;
    String label;
    String country;
    String format;
    int trackCount;
    boolean limitedEdition;

    public VinylRecord(String title, String artist, int year, String genre, String duration,
                       String label, String country, String format, int trackCount, boolean limitedEdition) {
        this.title = title;
        this.artist = artist;
        this.year = year;
        this.gener = genre;
        this.duration = duration;
        this.label = label;
        this.country = country;
        this.format = format;
        this.trackCount = trackCount;
        this.limitedEdition = limitedEdition;
    }

    public String toJson() {
        return String.format("""
        {
            "Insert title": "%s",
            "Insert artist": "%s",
            "Insert year": %d,
            "Insert gener": "%s",
            "Insert duration": "%s",
            "Insert label": "%s",
            "Insert country": "%s",
            "Insert format": "%s",
            "Insert trackCount": %d,
            "Insert limitedEdition": %s
        }
        """, title, artist, year, gener, duration, label, country, format, trackCount, limitedEdition);
    }
}

File: hw/fuentes/U2/HW10SaveJson/VinylRecord.py
import json

def registrerVinyl():
    vinyl_record = {
        "title": input("Insert title: "),
        "artist": input("Insert Artist: "),
        "year": int(input("Insert year: ")),
        "gener": input("Insert Gener: "),
        "timeline": input("Insert timeline (example. 47:23): "),
        "label": input("Insert label: "),
        "country": input("Insert country: "),
        "color": input("Insert color: "),
        "trackCount": int(input("Number of Tracks: ")),
        "LimitedEdition": input("¿Is a limited edition? (y/n): ").lower() == 'y'
    }

    with open("vinyl_record.json", "w", encoding="utf-8") as file:
        json.dump(vinyl_record, file, ensure_ascii=False, indent=4)

    print("✅ Registro guardado en vinyl_record.json")

registrerVinyl()